debezium-server-redis/src/main/java/io/debezium/server/redis/RedisMemoryThreshold.java

```diff
@@ -91,43 +114,55 @@ private Tuple2<Long, Long> memoryTuple(long defaultMaxMemory) {
         }
 
         Long usedMemory = parseLong(INFO_MEMORY_SECTION_USEDMEMORY, infoMemory.get(INFO_MEMORY_SECTION_USEDMEMORY));
-        if (usedMemory == null) {
-            return null;
-        }
-
-        Long maxMemory = parseLong(INFO_MEMORY_SECTION_MAXMEMORY, infoMemory.get(INFO_MEMORY_SECTION_MAXMEMORY));
-        if (maxMemory == null) {
-            if (defaultMaxMemory == 0) {
-                LOGGER.warn("Memory limit is disabled '{}'.", defaultMaxMemory);
-                return null;
+        Long configuredMemory = parseLong(INFO_MEMORY_SECTION_MAXMEMORY, infoMemory.get(INFO_MEMORY_SECTION_MAXMEMORY));
+        if (configuredMemory == null || (memoryLimit > 0 && configuredMemory > memoryLimit)) {
+            configuredMemory = memoryLimit;
+            if (configuredMemory > 0) {
+                LOGGER.debug("Setting maximum memory size {}", humanReadableSize(configuredMemory));
             }
-            LOGGER.debug("Using memory limit with value '{}'.", defaultMaxMemory);
-            maxMemory = defaultMaxMemory;
         }
-        else if (maxMemory == 0) {
-            LOGGER.debug("Redis 'info memory' field '{}' is {}. Consider configuring it.", INFO_MEMORY_SECTION_MAXMEMORY, maxMemory);
-            if (defaultMaxMemory > 0) {
-                maxMemory = defaultMaxMemory;
-                LOGGER.debug("Using memory limit with value '{}'.", defaultMaxMemory);
-            }
-        }
-
-        return Tuple2.of(usedMemory, maxMemory);
-    }
-
-    private void disable() {
-        isMemoryOk = MEMORY_OK;
-        LOGGER.warn("Memory threshold percentage check is disabled!");
+        return Tuple2.of(usedMemory, configuredMemory);
     }
 
     private Long parseLong(String name, String value) {
+        if (value == null) {
+            return null;
+        }
+
         try {
             return Long.valueOf(value);
         }
         catch (NumberFormatException e) {
             LOGGER.debug("Cannot parse Redis 'info memory' field '{}' with value '{}'.", name, value);
+            return null;
         }
-        return null;
     }
 
+    /**
+     * Formats a raw file size value into a human-readable string with appropriate units (B, KB, MB, GB).
+     *
+     * @param size The size value to be formatted.
+     * @return A human-readable string representing the file size with units (B, KB, MB, GB).
+     */
+    public static String humanReadableSize(Long size) {
+        if (size == null) {
+            return "Not configured";
+        }
+
+        final String[] units = { "B", "KB", "MB", "GB" };
+        int unitIndex = 0;
+
+        double sizeInUnit = size;
+
+        if (size == 0) {
+            return "0 B";
+        }
+
+        while (sizeInUnit >= 1024 && unitIndex < units.length - 1) {
+            sizeInUnit /= 1024.0;
+            unitIndex++;
+        }
+
+        return String.format("%.2f %s", sizeInUnit, units[unitIndex]);
+    }
```

Comment:
I'd suggest creating a follow-up Jira and consider moving this to `debezium-core` inside either the `io.debezium.util.NumberConversions` class or `io.debezium.util.Strings` utility classes as I think this could be helpful for any connector that may want to log similar bits in the future, perhaps with a variant that uses the `"%.2f %s" format by default and another that takes a format string that allows customization.

Reply:
I'd suggest to name this with a verb, something like:
- formatReadableSize
- getSizeInHumanReadableFormat
- convertSizeToReadableString
- getFormattedSizeString

Reply:
Changing to getSizeInHumanReadableFormat. I will create a new Jira for that.  This fix will need to go into different repository